modules/parsers/parser-hdbschema/com.sap.xsk.parser.hdbschema/src/main/java/com/sap/xsk/parser/hdbschema/models/XSKHDBSCHEMADefinitionModel.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2019-2021 SAP SE or an SAP affiliate company and XSK contributors
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-FileCopyrightText: 2019-2021 SAP SE or an SAP affiliate company and XSK contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.sap.xsk.parser.hdbschema.models;
+
+import com.sap.xsk.parser.hdbschema.exceptions.XSKHDBSchemaMissingPropertyException;
+import java.util.List;
+import java.util.Objects;
+
+public class XSKHDBSCHEMADefinitionModel {
+
+  private String schema;
+
+  public String getSchema() {
+    return schema;
+  }
+
+  public void setSchema(String schema) {
+    this.schema = schema;
+  }
+
+  public void checkForAllMandatoryFieldsPresence() throws Exception {
```

Comment:
You do not need this method, because your grammar will have only one property and if it is missing the g4 will not parse it.
 Thsi method is in good used if in your grammar you can not check if a field is mandatory in this case for example 
`property: schemaProp | publicProp | queryProp | dependsOnProp | dependsOnTable | dependsOnView;`
here the grammar just says that one of these fileds need to be set, and that is why we need an additional method to check for the mandatory.

If you need this method only for testing purposes, then it must be extracted in separate Util class.
The entity must be assembled in a way that it encapsulate only basic logic, not utils methods